### DISCUSSION TITLE
A likely bug of optimizer

### DISCUSSION THREAD

**Discussion by bakey** [duckdb/duckdb#10349] [General]
Hi, community.  I am seeking a help to idenfity is this code is a bug?
In the source : src/optimizer/filter_combiner.cpp, this function:
`FilterResult FilterCombiner::AddBoundComparisonFilter`
`bool left_is_scalar = comparison.left->IsFoldable();
  bool right_is_scalar = comparison.right->IsFoldable();`
The variable name is **_is_scalar, but is to call the isFoldable function. Does that call wrong function?

**Comment by royguo** [comment]
@bakey Could you explain a little bit more about this bug? For example, what is the current problem and what is the potential solution, thanks.

**Comment by Mytherin** [comment]
That's not a bug, no, although the variable name could perhaps be more clear. Foldable essentially means "scalar and has no side effects", i.e. `pi()` and `random()` are both scalars, but only `pi()` is foldable.

**Reply by bakey** [reply]
@Mytherin Thanks very much for your reply. Can you help to see my question that details here ? https://github.com/duckdb/duckdb/issues/10351
I try to replace IsFoldable to IsScalar, I can get a more optimize physical plan. If this code was correct , how do I get the most optimize plan ?
